---
title: "Genome Assembly Gaps"
author: "Ronica K"
output: 
  html_document:
    keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=8, dpi=300, echo=F, warning=F, message=F, cache=F)
```

#Mouse Genome Assemblies

downloaded from UCSC

```{r}
library(BSgenome.Mmusculus.UCSC.mm10)
library(ggplot2); library(RColorBrewer) 
library(reshape2)

source('../txtToGR.R')

filenames <- c('../GenomeTables/mm10gaps.txt.gz', '../GenomeTables/mm9gaps.txt.gz',
               '../GenomeTables/mm8gaps.txt.gz', '../GenomeTables/mm7gaps.txt.gz')
mmnames <- paste0('chr', c(1:19, 'X', 'Y'))

#21
mm10gaps <- GapstoGR(filenames[1], 'mm10', mmnames)
mm9gaps <- GapstoGR(filenames[2], 'mm9', mmnames)
mm8gaps <- GapstoGR(filenames[3], 'mm8', mmnames)
mm7gaps <- GapstoGR(filenames[4], 'mm7', mmnames)
```

##GRCm38/mm10 (Dec. 2011)
```{r}
sort(mm10gaps)
```

###Gap Types

__clone__ - gaps between clones

__contig__ - gaps between map contigs, various sizes

__telomere__ - 42 gaps for telomeres (100,000 Ns)

__centromere__ - 20 gaps for centromeres (size: 2,890,000 Ns)

__short_arm__ - 21 gaps for the short arm (10,000 Ns) at base positions 100,001-110,000 of each chromosome

__other__ - sequence of Ns in the assembly that were not marked as gaps in the AGP assembly definition file, various sizes

__fragment__ - a single gap of 31 bases in chrX_GL456233_random


```{r}
table(mm10gaps$type)
```


##NCBI37/mm9 (July 2007)
```{r}
sort(mm9gaps)
```

###Gap Types

__fragment__ - gaps between the contigs of a draft clone

__contig__ - gaps between map contigs

__centromere__ - gaps from centromeres (3,000,000 Ns) or other large blocks of heterochromatin (size varies)

```{r}
table(mm9gaps$type)
```


##NCBI36/mm8 (Feb. 2006)
```{r}
sort(mm8gaps)
```

###Gap Types

__fragment__ - gaps between the contigs of a draft clone

__contig__ - gaps between map contigs

__centromere__ - gaps from centromeres (3,000,000 Ns) or other large blocks of heterochromatin (size varies)


```{r}
table(mm8gaps$type)
```


##NCBI35/mm7 (Aug. 2005)
```{r}
sort(mm7gaps)
```

###Gap Types

__fragment__ - gaps between the contigs of a draft clone

__clone__ - gaps between clones in the same map contig

__contig__ - gaps between map contigs

__centromere__ - gaps from centromeres (3,000,000 Ns) or other large blocks of heterochromatin (size varies)


```{r}
table(mm7gaps$type)
```

##Gaps on Chromosome Y

__mm10__

```{r}
sort(mm10gaps[seqnames(mm10gaps)=='chrY'])
```

__mm9__

```{r}
sort(mm9gaps[seqnames(mm9gaps)=='chrY'])
```

__mm8__

```{r}
sort(mm8gaps[seqnames(mm8gaps)=='chrY'])
```

__mm7__

```{r}
sort(mm7gaps[seqnames(mm7gaps)=='chrY'])
```